data-raw/algorithm.csv

```diff
@@ -1,4 +1,10 @@
-name,logic
-hba1c,(analysiscode == 'NPU27300' AND value >= 48) OR (analysiscode == 'NPU03835' AND value >= 6.5)
-gld,atc =~ '^A10'
-
+register,name,title,logic,comments
+lab_forsker,hba1c,HbA1c inclusion,(analysiscode == 'NPU27300' AND value >= 48) OR (analysiscode == 'NPU03835' AND value >= 6.5),Is the IFCC units for NPU27300 and DCCT units for NPU03835
+lmdb,gld,Glucose-lowering drug inclusion,atc =~ '^A10' AND !(atc =~ '^(A10BJ|A10BK01|A10BK03)'),Do not keep GLP-RAs or dapagliflozin/empagliflozin drugs
+lpr_diag,lpr2,Relevant LPR2 diagnoses codes,c_diag =~ '^(DO0[0-6]|DO8[0-4]|DZ3[37]|DE1[0-4]|249|250)' AND (c_diagtype == 'A' OR c_diagtype == 'B'),'A' c_diagtype means primary diagnosis
+lpr_diag,lpr2_has_t1d,LPR2 diagnoses codes for T1D,c_diag =~ '^(DE10|249)',
+lpr_diag,lpr2_has_t2d,LPR2 diagnoses codes for T2D,c_diag =~ '^(DE11|250)',
+lpr_diag,lpr2_has_pregnancy_event,LPR2 diagnoses codes for pregnancy-related outcomes,c_diag =~ '^(DO0[0-6]|DO8[0-4]|DZ3[37])',These are recorded pregnancy endings like live births and miscarriages
+lpr_adm,lpr2_is_endocrinology_department,LPR2 endocrinology department,"na_if(c_spec, !(c_spec %in% 8:30)) == 8","True is when the department is endocrinology, false is when it is other medical departments, and missing is all other cases."
```

Comment:
When I was working with the simulated register data, I noticed that If `c_spec` is 1, this expression returns `NA`:
<img width="308" alt="image" src="https://github.com/user-attachments/assets/8548302d-d588-4886-8b82-22ca8cc807d7" />

Shouldn't it return `FALSE`?

Is it even possible that `c_spec` = 1 in the actual register data? If yes, we should probably change this expression. If not, it should be changed in the simulated data instead :)

Reply:
That's correct, it should give NA. The output should be if it is 8 then true, if it is 9 to 30, it is false, and all other cases, missing. Since we don't need those other cases when `c_spec` isn't between 8 to 30.

Reply:
Ahh, alrighty. That's great then! I must have misunderstood the text about this.